vermouth/gmx/topology.py

```diff
@@ -0,0 +1,216 @@
+                itp_file.write(f"{atype} {mass} {charge} A {nb1:3.8F} {nb2:3.8F} {comments}\n")
+
+def write_nonbond_params(system, itp_path, C6C12=False):
+    """
+    Writes the [nonbond_params] directive to file.
+    All atomtypes are defined in system.gmx_topology_params.
+    Masses and further information are taken from the molecule
+    directly.
+    """
+    conditional_keys = {True: '#ifdef', False: '#ifndef'}
+    with deferred_open(itp_path, "w") as itp_file:
+        itp_file.write("[ nonbond_params ]\n")
+        grouped_types = _group_by_conditionals(system.gmx_topology_params['nonbond_params'])
+        for (conditional, group), interactions_in_group in grouped_types:
+            if conditional:
+                conditional_key = conditional_keys[conditional[1]]
+                itp_file.write('{} {}\n'.format(conditional_key, conditional[0]))
+            if group:
+                itp_file.write('; {}\n'.format(group))
+
+
+            for nb_params in interactions_in_group:
+                if len(nb_params.atoms) == 2:
+                    a1, a2 = nb_params.atoms
+                # self interaction
+                else:
+                    a1 = nb_params.atoms[0]
+                    a2 = nb_params.atoms[0]
+
+                if C6C12:
+                    nb1, nb2 = convert_sigma_epsilon(nb_params.sigma, nb_params.epsilon)
+                else:
+                    nb1, nb2 = nb_params.sigma, nb_params.epsilon
+
+                comments = ";" + " ".join(nb_params.meta['comment'])
+                itp_file.write(f"{a1} {a2} 1 {nb1:3.8F} {nb2:3.8F} {comments}\n")
+
+def write_gmx_topology(system, top_path, itp_paths=[], C6C12=False, defines=(), header=()):
+    """
+    Writes a Gromacs .top file for the specified system. Gromacs topology
+    files are defined by directives for example `[ atomtypes ]`. However,
+    Gromacs supports writing parts of the topology to so called .itp
+    files which can be inculded into a toplevel topology file with the
+    extension .top using #include statements. The topology writer will
+    generate such a toplevel topology file where the different directives
+    are written to seperate .itp files and included into the toplevel
+    file.
+
+    Parameters
+    ----------
+    system: :class:vermouth.System
+    top_path: pathlib.Path
+        path for topology file
+    defines: tuple(str)
+        define statments to include in the topology
+    header: tuple(str)
+        any comment lines to include at the beginning
+    """
+    if not system.molecules:
+        raise ValueError("No molecule in the system. Nothing to write.")
+
+    include_string = ""
+    # First we write the atomtypes and nonbondparams directive
+    if itp_paths:
+        write_atomtypes(system, itp_paths[0], C6C12)
+        write_nonbond_params(system, itp_paths[1], C6C12)
+        include_string = "\n".join('#include "{}"'.format(path) for path in itp_paths[:2])
+        include_string += "\n"
```

Comment:
Not sure I like this. Does it make more sense to do e.g. `if 'atomtypes' in system.gmx_topology_params`?